hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/PipelineManagerV2Impl.java

```diff
@@ -111,15 +113,25 @@ private PipelineManagerV2Impl(ConfigurationSource conf,
     this.pipelineCreationAllowed = new AtomicBoolean(!this.isInSafeMode.get());
   }
 
-  public static PipelineManagerV2Impl newPipelineManager(
+  public static PipelineManagerV2Impl newPipelineManagerWithMockBuffer(
```

Comment:
`DBTransactionBuffer` can be fetched by calling `SCMHAManager#getDBTransactionBuffer`, so this function `newPipelineManagerWithMockBuffer` is not needed.
```
// Create PipelineStateManager
StateManager stateManager = PipelineStateManagerV2Impl
    .newBuilder().setPipelineStore(pipelineStore)
    .setRatisServer(scmhaManager.getRatisServer())
    .setNodeManager(nodeManager)
    .setSCMDBTransactionBuffer(scmhaManager.getDBTransactionBuffer())
    .build()
```

Reply:
This is a very good point.